source/keyboardHandler.py

```diff
@@ -387,6 +387,11 @@ def _get_shouldReportAsCommand(self):
 			# This could be for an event such as gyroscope movement,
 			# so don't report it.
 			return False
+		if self.vkCode in self.TOGGLE_KEYS:
+			# #5490: Dont report for keys that toggle on off.
+			# This is to avoid them from reported twice: once by the 'speak command keys' feature,
+			# and once by the 'speak typed characters' feature

```

Comment:
Just as a tiny clarification, this isn't affected in any way by "speak typed characters". That is, "caps lock on", etc. is always spoken, even if speak typed characters is off.